runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -881,13 +881,51 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, job
 					vmindex = (jlong)(UDATA)methodID;
 					target = (jlong)(UDATA)method;
 
+					J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methodID->method);
+					J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethod);
 					new_clazz = J9VM_J9CLASS_TO_HEAPCLASS(J9_CLASS_FROM_METHOD(method));
+					new_flags = romMethod->modifiers & CFR_METHOD_ACCESS_MASK;
 
-					J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methodID->method);
-					new_flags = flags | (romMethod->modifiers & CFR_METHOD_ACCESS_MASK);
 					if (J9_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccMethodCallerSensitive)) {
 						new_flags |= MN_CALLER_SENSITIVE;
 					}
+
+					if (J9_ARE_ALL_BITS_SET(flags, MN_IS_METHOD)) {
+						new_flags |= MN_IS_METHOD;
+						if (MH_REF_INVOKEINTERFACE == ref_kind) {
```

Comment:
should there be an assert for `Assert_JCL_true(J9_ARE_NO_BITS_SET(romMethod->modifiers , J9AccStatic));` here as well

Reply:
fixed